frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/PaymentView.tsx

```diff
@@ -75,36 +84,93 @@ export const PaymentView = () => {
     setFieldBuilderToInactive()
     setDesignState(DesignState.Inactive)
 
-    handleClose(false)
-    handleBuilderClick(false)
+    if (!isMobile) handleBuilderClick(false)
     setFieldListTabIndex(FieldListTabIndex.Payments)
   }
 
-  return paymentDetails.enabled ? (
-    <Box w="100%" maxW="57rem" alignSelf="center" ref={paymentRef}>
-      <FormProvider {...formMethods}>
-        <Box mt="2.5rem" bg="white" py="2.5rem" px="1.5rem">
-          <Box
-            transition="background 0.2s ease"
-            _hover={{ bg: 'secondary.100', cursor: 'pointer' }}
-            borderRadius="4px"
-            _active={{
-              bg: 'secondary.100',
-              boxShadow: '0 0 0 2px var(--chakra-colors-primary-500)',
-            }}
-            data-active={isActive || undefined}
-            onClick={handlePaymentClick}
-          >
-            <Box p={{ base: '0.75rem', md: '1.5rem' }}>
-              <PaymentPreview
-                colorTheme={form?.startPage.colorTheme}
-                paymentDetails={paymentDetails}
-                isBuilder
-              />
+  return (
+    (paymentDetails.enabled ||
+      paymentState === PaymentState.EditingPayment) && (
```

Comment:
Would prefer to use guard clauses here. This should help to reduce the indents and cognitive load as we can exclude unhappy cases earlier.
```
if (!paymentDetails.enabled && paymentState !== PaymentState.EditingPayment) { 
    return null
}

return (<Box>{...}</Box>)
```

Reply:
[f5f41ce](https://github.com/opengovsg/FormSG/pull/6511/commits/f5f41cea39fa3d7eeb0354169637fd99dc1d551b)!